kitsune/wiki/views.py

```diff
@@ -1717,7 +1717,7 @@ def recent_revisions(request):
     c = {
         "revisions": revs,
         "form": form,
-        "locale": request.GET.get("locale", request.LANGUAGE_CODE),
+        "locale": locale or request.LANGUAGE_CODE,  # Ensure locale is always set
```

Comment:
I don't understand this change as well. Why are you no longer getting the locale from the URL's query parameters?

Reply:
What basically this does is equivalent to `"locale": request.LANGUAGE_CODE`. You are setting `locale=None` which will force the short circuit operation to evaluate always to `request.LANGUAGE_CODE`

The problem here is the get() method. It defaults to `request.LANGUAGE_CODE` if the "locale" dictionary key is not present. It doesn't check the value of the key which is the empty string. 

This should be:
```python
"locale": request.GET.get("locale") or request.LANGUAGE_CODE

Reply:
You are right of course; correcting.